### PR TITLE
Sonar warnings

### PR DESCRIPTION
**Pull request by wcekan**
* Never ignore InterruptedException
* Use immutable collections where possible
* Use proper immutable collectors

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/KeyValueStoreDimension.java [hunk 1]
```diff
@@ -475,7 +475,7 @@ public void addAllDimensionRows(Set<DimensionRow> dimensionRows) {
                 indexRows.put(rowIdKey, new Pair<>(dimensionRow, dimensionRowOld));
 
             } catch (IOException e) {
-                LOG.error("Exception while adding dimension row {}", e);
+                LOG.error("Exception while adding dimension row", e);
```
Comment (wcekan) [inline fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/KeyValueStoreDimension.java:478]: https://sonarcloud.io/project/issues?id=yahoo_fili&issues=AXPLWYhWIUmnqh_n8dv8&open=AXPLWYhWIUmnqh_n8dv8
> Not enough arguments

When the final argument is a Throwable it generates a stack trace and it is not used for formatting.  This maintains current functionality, but was original intent to show just the exception?
```java
LOG.error("Exception while adding dimension row {}", e.toString());
```

Comment (michael-mclawhorn) [reply fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/KeyValueStoreDimension.java:478]: The original intent was to make sure we didn't lose the precise logging associated with the error.
Stack trace is a very nice to have.

File: fili-core/src/test/java/com/yahoo/bard/webservice/web/endpoints/TestFilterServlet.java [hunk 2]
```diff
@@ -124,7 +124,7 @@ public void write(OutputStream os) throws IOException {
                 try {
                     Thread.sleep(100);
                 } catch (InterruptedException ignore) {
-                    // Ignore
+                    Thread.currentThread().interrupt();
```
Comment (michael-mclawhorn) [inline fili-core/src/test/java/com/yahoo/bard/webservice/web/endpoints/TestFilterServlet.java:127]: 👍

**Review by michael-mclawhorn — APPROVED** [review]

**Comment by michael-mclawhorn** [comment]
Could use a changelog for clarity.

**Pull request merged**